wxQt RTL fixes

AliKet opened the pull request:
### Some screenshots of the result of this PR:

**Widgets demo:**
<img width="1366" height="741" alt="widgets_rtl" src="https://github.com/user-attachments/assets/8f803fb0-6f15-4dfa-ae06-a16e6035cd05" />

**Grid demo:**
<img width="800" height="705" alt="griddemo_rtl" src="https://github.com/user-attachments/assets/b67a11ec-72d0-4ded-ae88-9d313a6db810" />

**AUI demo:**
<img width="801" height="631" alt="auidemo_rtl" src="https://github.com/user-attachments/assets/fced2ecc-4bed-4d44-a32e-e1c09c656d27" />

Commit:
wxQt: Fix window positioning in RTL layout.

We need to position the windows manually, because QWidget::setLayoutDirection()
only takes effect if the widget is managed by a QLayout, which we don't use in
wxWidgets.

We also need to reposition child windows if the parent client size is changed,
as Qt does nothing in this case.

Commit:
wxQt: Fix wxWindow::Refresh() not working correctly in RTL layout.

Commit:
wxQt: Fix window scrolling in RTL layout

Commit:
wxQt: Fix ClientToScreen() and ScreenToClient() to work correctly in RTL layout.

In RTL layout, wxWidgets expects the logical point (0,0) to be in the top right
corner of the window, while Qt expects it to always be in the top left corner.

Commit:
wxQt: Fix popup menu position in RTL layout

Commit:
wxQt: Override wxDC::{Get,Set}LayoutDirection()

Commit:
wxQt: Apply custom mirroring to wxDC in RTL layout.

Commit:
wxQt: Fix wxDC::GetSize() for wxMemoryDC

Commit:
wxQt: Fix drawing wxGrid lines in RTL layout

Commit:
wxQt: Fix drawing wxTreeCtrl icons in RTL layout

Commit:
wxQt: Fix wxOverlay in RTL layout

Commit:
wxQt: Fix rendering some native control in RTL layout

Commit:
wxQt: Override wxDC::SetLayoutDirection() in wxMemoryDC

For correct drawing in RTL layout, because the base class version doesn't
(and shouldn't) mirror the DC by default.

vadz reviewed (commented):
Thanks, this looks good and could be merged AFAICS, I'm not sure why is this in the "draft" state yet?

Also, unfortunately I have no idea why did `SliderTestCase` start consistently failing with Qt 6.8 just recently, but this is  unrelated to this change.

Commit:
wxQt: Simplify wxDC::DoDraw{Rotated}Text() implementations

No real change, just a simplification to make them easier to fix in the RTL layout.

Commit:
wxQt: Fix drawing multiline text with wxDC::DrawText()

Commit:
wxQt: Fix drawing (rotated) text in RTL layout

Commit:
wxQt: Fix drawing (rotated) text with GC in RTL layout

AliKet commented on src/aui/framemanager.cpp:
```diff
@@ -3478,11 +3478,13 @@ wxRect wxAuiManager::CalculateHintRect(wxWindow* pane_window,
 
     m_frame->ClientToScreen(&rect.x, &rect.y);
 
+#ifndef __WXQT__
```
This code is also not needed when this [PR](#25426) is complete. see this [commit](https://github.com/wxWidgets/wxWidgets/pull/25426/commits/ee9c8e01197de157eb304006a4a8928dae210897)

AliKet commented on src/qt/overlay.cpp:
```diff
@@ -100,10 +108,14 @@ void wxOverlayImpl::Init(wxDC* dc, int , int , int , int )
 {
     wxCHECK_RET( IsOk(), "wxOverlay not initialized" );
 
+    // For correct result in RTL layout, the drawing is done in a non-mirrored
+    // DC and the mirroring is done in paintEvent() above.
```
If we don't do this (i.e.: forcing LTR on the DC) the rubberbanding is not drawn correctly, i.e.: the rubberband is missing near the right side of the window (image 01) and is drawn over the vertical scrollbar (image 02):

Image 01:
<img width="800" height="730" alt="drawing_1" src="https://github.com/user-attachments/assets/bcc6c84a-d12c-4d4f-abfd-c9e23ea8ecbf" />

Image 02:
<img width="800" height="730" alt="drawing_2" src="https://github.com/user-attachments/assets/c3c76a88-2810-4b06-9c5a-62ceb0e20a08" />

I haven't found the reason for this problem. I'll add a comment stating that this is to correct drawing the rubberband on a scrolled window.

AliKet commented on tests/graphics/clippingbox.cpp:
```diff
@@ -1033,9 +1033,9 @@ static void OneDevRegionRTL(wxDC& dc, const wxBitmap& bmp, bool useTransformMatr
     }
 
-#ifdef __WXGTK__
+#if defined(__WXGTK__) || defined(__WXMSW__)
```
The test is enabled for both `wxMSW` and `wxGTK3` in this [PR](#25426) (see [4f49395](https://github.com/wxWidgets/wxWidgets/pull/25426/commits/4f49395d979303efafea9661a420719a633a3b5d) and [403e9f2](https://github.com/wxWidgets/wxWidgets/pull/25426/commits/403e9f28d977d1c2c9c96498e53c45ee6f513abe))

I think I should merge the changes done here in that PR and close this one! what do you think ?

vadz replied:
Yes, probably, but whatever you decide please let me know how do you want me to process them, i.e. in which order to merge them if you don't combine them both. TIA!

AliKet commented on src/qt/dc.cpp:
```diff
@@ -462,10 +496,10 @@ void wxQtDCImpl::Clear()
-                            DeviceToLogicalY(0),
-                            DeviceToLogicalXRel(width),
-                            DeviceToLogicalYRel(height) );
+    const wxPoint pos = DeviceToLogical(0, 0);
```
`DeviceToLogical{X,Y,...}()` return different values ​​than their counterpart in the other ports (i.e.: `wxMSW` and `wxGTK3`), which makes `OneDevRegionRTL` test case in `wxQt` fail!

vadz replied:
Hmm, but which one is correct?

Commit:
wxQt: Fix SetClientSize() not working correctly in RTL layout

Commit:
wxAUI: Don't mirror hint rectangle in RTL layout

 This is for wxQt.

Commit:
Relax the clip box check in OneDevRegionRTL test case

Due to rounding errors, this test would fail under wxMSW if the region started
at x=2*n, and would fail under wxQt if it started at x=2*n+1. So relax the check
by using 1 as a position tolerance.

AliKet commented:
@vadz  Should I completely remove the code in [55f5433](https://github.com/wxWidgets/wxWidgets/pull/25822/commits/55f54331c94e11df7f75dd0cd57d674a05997041) because it seems unnecessary at least under `wxMSW`, `wxGTK3` and `wxQt`?

AliKet commented:
> Ideal would be to check if this code is needed under Mac and, if it is, and assuming the bug can't be fixed in wxOSX itself, replace the preprocessor condition with a check for `__WXOSX__`. TIA!

I don't have a Mac device to test this on it. so I'll leave the check as is then.

vadz commented:
FWIW we have a wxMac which is specifically there to be used for testing by people without access to a Mac. Please contact Stefan to get access to it.

In the meanwhile, let me know what do I need to do to test for it myself and I could do it here.

vadz committed:
Remove probably unnecessary RTL workaround from wxAuiManager

This RTL workaround is not needed for wxMSW, wxGTK and wxQt and wxOSX
doesn't seem to support RTL anyhow, so it's highly likely that it's not
really needed anywhere, hence just remove it.
